pkg/gardenadm/cmd/init/init.go

```diff
@@ -148,6 +150,18 @@ func run(ctx context.Context, opts *Options) error {
 			Fn:           b.Shoot.Components.SystemComponents.Namespaces.Wait,
 			Dependencies: flow.NewTaskIDs(waitUntilGardenerResourceManagerReady, deployShootNamespaces),
 		})
+		_ = g.Add(flow.Task{
+			Name:         "Deploying kube-proxy system component",
+			Fn:           b.DeployKubeProxy,
+			SkipIf:       !kubeProxyEnabled,
+			Dependencies: flow.NewTaskIDs(waitUntilGardenerResourceManagerReady, waitUntilShootNamespacesReady, waitForExtensionControllersReady),
+		})
+		_ = g.Add(flow.Task{
+			Name:         "Deleting kube-proxy system component",
+			Fn:           b.Shoot.Components.SystemComponents.KubeProxy.Destroy,
+			SkipIf:       kubeProxyEnabled,
+			Dependencies: flow.NewTaskIDs(waitUntilGardenerResourceManagerReady),
+		})
```

Comment:
Is this just being a good citizen and including the clean-up as well or is there more to it? I would have hoped nobody changes there mind about `kube-proxy` deployment half-way through `gardenadm init`. Day 2 lifecycle management should not be performed by `gardenadm`.
That being said, I do not mind this single task and was just wondering if we changed directions meanwhile.